src/number_theory/bernoulli.lean

```diff
@@ -258,3 +265,35 @@ begin
   { ext x, rw bernoulli_eq_bernoulli' (succ_ne_zero x ∘ succ.inj) },
   { ring },
 end
+
+theorem bernoulli_power_series :
+  power_series.mk (λ n, (bernoulli n / n! : ℚ)) * (exp ℚ - 1) = X :=
```

Comment:
In the conversation on #6409, @eric-wieser suggested making the statements of the power series lemmas for an arbitrary `ℚ`-algebra `A`, not just `ℚ` itself.  This seems like a good idea to me (you could do it above for `bernoulli'_power_series`, too).

Also, I don't really know the power series part of the library, but I'd be tempted to move the power series lemmas to `ring_theory/power_series`, either in `ring_theory/power_series/well_known` or in a new file `ring_theory/power_series/bernoulli`, with the idea that they might be easier to find there ... @jcommelin what do you think?

Reply:
> In the conversation on #6409, @eric-wieser suggested making the statements of the power series lemmas for an arbitrary (commutative?) `ℚ`-algebra `A`, not just `ℚ` itself. This seems like a good idea to me (you could do it above for `bernoulli'_power_series`, too).
> 
Seems fine to me, what would be the exact statement we want? This does not seem to work:
```
variables {A : Type*} [comm_ring A]

theorem bernoulli'_power_series  [algebra ℚ A]:
  power_series.mk (λ n, (bernoulli' n / n! : A)) * (exp A - 1) = X * exp A :=
```


> Also, I don't really know the power series part of the library, but I'd be tempted to move the power series lemmas to `ring_theory/power_series`, either in `ring_theory/power_series/well_known` or in a new file `ring_theory/power_series/bernoulli`, with the idea that they might be easier to find there ... @jcommelin what do you think?
> 
> (It would not occur to me to look in `number_theory/*` to find the formal power series inverse of `(e ^ t - 1) / t` ;-). Evidently I'm not a number theorist!)

I'm fine to put it wherever, just let me know!

Reply:
> Seems fine to me, what would be the exact statement we want? This does not seem to work:
> 
> ```
> variables {A : Type*} [comm_ring A]
> 
> theorem bernoulli'_power_series  [algebra ℚ A]:
>   power_series.mk (λ n, (bernoulli' n / n! : A)) * (exp A - 1) = X * exp A :=
> ```

One needs an `algebra_map ℚ A`.  I just pushed the `bernoulli'_power_series` version as an example; if you like it, you can do the `bernoulli_power_series` lemma in the same way!

Reply:
Thanks for the example! I adapted the proof for `bernoulli_power_series` to include the `algebra_map`. However I couldn't get the derivation from `bernoulli'_power_series` to work for `ring A` but only for `integral domain A`. This is because in the derivation, we use `mul_right_cancel'` and  also assume that `-1 \ne 0` in `A`. Perhaps we can get around it somehow, but I guess "Q algebra over integral domain" is already better than Q only.

Reply:
Are you sure you need integral_domain and not just domain?

Reply:
At least the current proof does not fly without it. The statement of `bernoulli_power_series` is probably true as in `bernoulli'_power_series` for `ring`. Perhaps we should aim at a proof that does not use the reduction to `bernoulli'_power_series`.

Reply:
Completed now the proof above for `comm_ring`, except for the lemma `bernoulli_spec'`, which should follow easily from `bernoulli_spec`...

Reply:
> At least the current proof does not fly without it. The statement of `bernoulli_power_series` is probably true as in `bernoulli'_power_series` for `comm_ring`. Perhaps we should aim at a proof that does not use the reduction to `bernoulli'_power_series`?

I don't think we can prove `bernoulli'_power_series` without `integral_domain A`. But if it can be done for `bernoulli_power_series`, then it can surely be done for `bernoulli'_power_series` as well (since they are just image and preimage under an injective map)?

Reply:
Also, as far as the statement of `bernoulli'_power_series` for `A` is concerned, shouldn't it just be requiring a coercion, with something like `(exp \Q) : power_series A` is the same as `exp A`?

Reply:
> > At least the current proof does not fly without it. The statement of `bernoulli_power_series` is probably true as in `bernoulli'_power_series` for `comm_ring`. Perhaps we should aim at a proof that does not use the reduction to `bernoulli'_power_series`?
> 
> I don't think we can prove `bernoulli'_power_series` without `integral_domain A`. But if it can be done for `bernoulli_power_series`, then it can surely be done for `bernoulli'_power_series` as well (since they are just image and preimage under an injective map)?

Heather provided a proof for `bernoulli'_power_series` with 'comm_ring A`.

Reply:
Ok, now the pull request contains 

1. A modified version of `bernoulli'_power_series` with `exp A` for `comm_ring A` (thanks @hrmacbeth!)
2. A version of `bernoulli_power_series`with `exp A` for `comm_ring A`. This proof does not use `bernoulli'_power_series`. It is similar to the proof of `bernoulli'_power_series`, which relies on `bernoulli'_spec'`. We use instead:
3. `bernoulli_spec'`: similar to `bernoulli'_spec'` (hence the name), and we basically reduce to `bernoulli'_spec'` in the proof. 

It all seems to work, but the following things might still need improvement:
- The proof of 2. has some lines that are identical to 1., don't know if this is a problem.
- The proof of 3. is currently a bit long.

Reply:
Thanks!  I'll make a pass and see if I can shorten some things.  Comments will trickle in bit by bit.